### ISSUE TITLE
Incorrect aggregate job icon

### ISSUE THREAD

**Issue by wxtim** [cylc/cylc-ui#1098]
**Describe the bug**
With a workflow where it takes multiple tries to get a task to pass.

With the second try the aggregate icon doesn't seem to show the state of the first try, but reflect the state of the second try.
At the third and subsequent retry this works as expected.

![cylc-ui-task-state-bug](https://user-images.githubusercontent.com/26465611/184608116-46101808-ddbb-4e6f-b5a1-3e473fdebed6.gif)

**Steps to reproduce the bug**

I used this workflow

```
[scheduler]
    allow implicit tasks = true

[scheduling]
    initial cycle point = 1066
    [[graph]]
        PT1H = food => barn => brash => quotidian

[runtime]
    [[barn]]
        script = test "${CYLC_TASK_SUBMIT_NUMBER}" == 4 || exit 1 && exit 0

```
**Additional context**

Discovered whilst investigating 

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by MetRonnie** [comment]
Duplicate of #1099

**Comment by wxtim** [comment]
I have no idea how I managed that!
